pkg/cmd/pulumi/util.go

```diff
@@ -161,18 +161,29 @@ func commandContext() context.Context {
 
 func createSecretsManager(
 	ctx context.Context, stack backend.Stack, secretsProvider string,
-	rotatePassphraseSecretsProvider bool) error {
-	configFile, err := getProjectStackPath(stack)
-	if err != nil {
-		return err
-	}
+	rotatePassphraseSecretsProvider, creatingStack bool) error {
 
 	// As part of creating the stack, we also need to configure the secrets provider for the stack.
 	// We need to do this configuration step for cases where we will be using with the passphrase
 	// secrets provider or one of the cloud-backed secrets providers.  We do not need to do this
 	// for the Pulumi service backend secrets provider.
 	// we have an explicit flag to rotate the secrets manager ONLY when it's a passphrase!
 	isDefaultSecretsProvider := secretsProvider == "" || secretsProvider == "default"
+
+	// If we're creating the stack, it's the default secrets provider, and it's the cloud backend
+	// return early to avoid probing for the project and stack config files, which otherwise
+	// would fail when creating a stack from a directory that does not have a project file.
+	if isDefaultSecretsProvider && creatingStack {
+		if _, isCloud := stack.Backend().(httpstate.Backend); isCloud {
```

Comment:
I've been trying to clean up these backend type checks, I suspect remote deployment is going to add a scattering of these back in all over the place :(

Reply:
I know, I'm sorry :( I was trying to avoid it, but had to do it in 2 places.

Here was kind of unavoidable without a larger refactoring.

The other place is in `runDeployment` in `util_remote.go`. I wanted this one to go away completely and have all of this live as an implementation detail inside the cloud backend, e.g. have the remote flags get passed-through to the `backend.Stack.Update()` call via `backend.UpdateOperation` -- but that ended up being a larger change in each of the commands, so I've avoided it for now.